### PR TITLE
CLN: simplify to_offset

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/tseries/frequencies/test_to_offset.py [hunk 1]
```diff
@@ -146,15 +147,6 @@ def test_to_offset_pd_timedelta(kwargs, expected):
     assert result == expected
 
 
-def test_to_offset_pd_timedelta_invalid():
-    # see gh-9064
-    msg = "Invalid frequency: 0 days 00:00:00"
-    td = Timedelta(microseconds=0)
-
-    with pytest.raises(ValueError, match=msg):
```
Comment (jreback) [inline pandas/tests/tseries/frequencies/test_to_offset.py:154]: is this duplicative?

Comment (jbrockmendel) [reply pandas/tests/tseries/frequencies/test_to_offset.py:154]: this no longer raises, is added as a case on L140

Comment (jreback) [reply pandas/tests/tseries/frequencies/test_to_offset.py:154]: ahh ok (oh the microseconds), sure

**Comment by jreback** [comment]
lgtm ping on green.

**Comment by jbrockmendel** [comment]
ping

**Pull request merged**